### ISSUE TITLE
Feature request: offline / disconnected mode 

### ISSUE THREAD

**Issue by kzorba** [cachix/devenv#1118]
I would like devenv to work in a disconnected environment (during flights for example). If we disconnect, here is an example of what we get:

```
# kzorba @ enigma in ~/WorkingArea/python-project on git:main x [12:43:37]
$ devenv version
devenv 1.0.1 (aarch64-darwin)

# kzorba @ enigma in ~/WorkingArea/python-project on git:main x [12:43:40]
$ devenv shell
• Building shell ...
thread 'main' panicked at devenv/src/command.rs:270:26:
Failed to get cache: reqwest::Error { kind: Request, url: Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("cachix.org")), port: None, path: "/api/v1/cache/nixpkgs-python", query: None, fragment: None }, source: hyper::Error(Connect, ConnectError("dns error", Custom { kind: Uncategorized, error: "failed to lookup address information: nodename nor servname provided, or not known" })) }
note: run with RUST_BACKTRACE=1 environment variable to display a backtrace
✔ Building shell in 1.6s.`` (edited)

```

**Comment by DaniruKun** [comment]
Second this, also train networks can be spotty, and this would be quite annoying.

**Comment by domenkozar** [comment]
Released in 1.0.4
